### PR TITLE
feat: APP-321 APP-389 add language switcher

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: APP-321 APP-389

Add language switcher.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@blushi @blushi You can check the language switcher here: https://deploy-preview-2515--regen-marketplace.netlify.app/projects/1

Note that I translated a small portion of the po file with a personal account to try out the feature but we'll have to translate the full file with a regen org account on an ai service.

**Comment by erikalogie** [comment]
Looks like it is working well though some of the translations need to be tweaked. Did you see my comment in the actual Jira issue about the styling @flagrede?

**Comment by flagrede** [comment]
@erikalogie I just updated the language switcher to match their design; it should be closer now.

**Comment by blushi** [comment]
@erikalogie do we have a design for the language switcher on our app?

**Review by blushi — COMMENTED** [review]
Looks good, but should we have a separate task for using the selected language in our sanity and server queries once data is translated there or should we handle the code changes here? I was assuming it could be tackled here as well since we don't even have all static text translated either.

**Comment by flagrede** [comment]
@blushi I can handle that in this PR.

**Comment by erikalogie** [comment]
> https://deploy-preview-2515--regen-marketplace.netlify.app/projects/1

Yes: https://www.figma.com/design/OXkWyviKwKCDT80HjzCbpE/Project-page-and-project-search-page?node-id=2628-210609&t=lj21Xni30azwI0mF-1 and https://www.figma.com/design/OXkWyviKwKCDT80HjzCbpE/Project-page-and-project-search-page?node-id=3234-22844&t=lj21Xni30azwI0mF-1

**Comment by erikalogie** [comment]
> @erikalogie do we have a design for the language switcher on our app?

Yes: https://www.figma.com/design/OXkWyviKwKCDT80HjzCbpE/Project-page-and-project-search-page?node-id=2628-210609&t=lj21Xni30azwI0mF-1 and https://www.figma.com/design/OXkWyviKwKCDT80HjzCbpE/Project-page-and-project-search-page?node-id=3234-22844&t=lj21Xni30azwI0mF-1

**Comment by flagrede** [comment]
@blushi I handled both sanity and graphql requests with the new language code.
I also added the language switcher to Regen client with the appropriate design.

**Comment by flagrede** [comment]
@blushi I have finished wiring everything for both sanity and graphql requests.
I have also setup some test data on staging:
Graphql translation (story field): https://deploy-preview-2515--regen-marketplace.netlify.app/project/C02-009
Sanity translation (credibility card): https://deploy-preview-2515--regen-marketplace.netlify.app/project/biocultural-jaguar-credits-ancestral-stewardship-in-the-sharamentsa-community-2

File: web-marketplace/sanity-graphql.schema.json [hunk 1]
```diff
@@ -45930,6 +45930,18 @@
             "isDeprecated": false,
             "deprecationReason": null
           },
+          {
+            "name": "complianceCredits",
```
Comment (blushi) [inline web-marketplace/sanity-graphql.schema.json:45934]: doesn't it to hurt but we should probably revert this for now

**Review by blushi — APPROVED** [review]
Great work!

**Comment by erikalogie** [comment]
@flagrede did you see that they put the language switcher on mobile into the main navigation bar? I've updated the comps to reflect: https://www.figma.com/design/brkxGV5qNOkZUp0YQl1cxO/Terrasos-Phase-1?node-id=2028-115990&t=DzUnWGMvsE1sPY0z-1

Also noting that their font-size on desktop for the language switcher is 15px

**Comment by flagrede** [comment]
@erikalogie I just updated to the latest design changes based on their website. Let me know if you have any feedback.

**Comment by flagrede** [comment]
@erikalogie I'll merge the PR as it's a big one to avoid conflicts. I can do a follow-up PR if needed.

**Comment by erikalogie** [comment]
It looks good!

**Pull request merged**